api/src/main/java/javax/enterprise/inject/spi/ProcessProducer.java

```diff
@@ -73,6 +77,21 @@
      * @throws IllegalStateException if called outside of the observer method invocation
      */
     public void setProducer(Producer<X> producer);
+    
+    /**
+     * Returns a {@link ProducerConfigurator} initialized with the {@link Producer} processed by this event, to configure a new
+     * {@link Producer} that will replace the original one at the end of the observer invocation.
+     * 
+     * <p>
+     * Each call returns the same configurator instance within an observer notification.
+     * </p>
+     * 
+     * @return a non reusable {@link ProducerConfigurator} to configure the original
+     *         {@link javax.enterprise.inject.spi.Producer}.
+     * @throws IllegalStateException if called outside of the observer method invocation
+     * @since 2.0
+     */
+    public ProducerConfigurator<X> configureProducer();
 
```

Comment:
for consistency with other Lifecycle events where we provide access to configurators, thus method should be renamed to `setProducer()`